clamp 1.3.0 default_XXX method behavior breaking fpm

It seems like clamp's new behavior for default methods (introduced in https://github.com/mdub/clamp/commit/ec192bf677ea518299dc16bf6733c00837a83b58 and released with version 1.3.0) breaks fpm with the following error:

```
/usr/local/lib/ruby/gems/2.4/gems/fpm-1.10.2/lib/fpm/command.rb:397:in `block in execute': undefined method `default_architecture' for #<FPM::Command:0x000000080462f870> (NoMethodError)
        from /usr/local/lib/ruby/gems/2.4/gems/fpm-1.10.2/lib/fpm/command.rb:403:in `execute'
        from /usr/local/lib/ruby/gems/2.4/gems/clamp-1.3.0/lib/clamp/command.rb:66:in `run'
        from /usr/local/lib/ruby/gems/2.4/gems/fpm-1.10.2/lib/fpm/command.rb:575:in `run'
        from /usr/local/lib/ruby/gems/2.4/gems/clamp-1.3.0/lib/clamp/command.rb:140:in `run'
        from /usr/local/lib/ruby/gems/2.4/gems/fpm-1.10.2/bin/fpm:7:in `<top (required)>'
        from /usr/local/bin/fpm:23:in `load'
        from /usr/local/bin/fpm:23:in `<main>'
```

This is still broken?
```
/Library/Ruby/Gems/2.3.0/gems/fpm-1.11.0/lib/fpm/command.rb:397:in `block in execute': undefined method `default_architecture' for #<FPM::Command:0x00007fe839979d88> (NoMethodError)
        from /Library/Ruby/Gems/2.3.0/gems/fpm-1.11.0/lib/fpm/command.rb:402:in `execute'
        from /Library/Ruby/Gems/2.3.0/gems/clamp-1.3.0/lib/clamp/command.rb:66:in `run'
        from /Library/Ruby/Gems/2.3.0/gems/fpm-1.11.0/lib/fpm/command.rb:574:in `run'
        from /Library/Ruby/Gems/2.3.0/gems/clamp-1.3.0/lib/clamp/command.rb:140:in `run'
        from /Library/Ruby/Gems/2.3.0/gems/fpm-1.11.0/bin/fpm:7:in `<top (required)>'
        from /usr/local/bin/fpm:22:in `load'
        from /usr/local/bin/fpm:22:in `<main>'
make: *** [jboss] Error 1


❯ fpm --version
1.11.0

❯ ruby --version
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]

❯ gem list | grep clamp
clamp (1.0.1)

❯ gem environment
RubyGems Environment:
  - RUBYGEMS VERSION: 3.0.3
  - RUBY VERSION: 2.6.3 (2019-04-16 patchlevel 62) [x86_64-darwin18]
  - INSTALLATION DIRECTORY: /Users/bedge/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0
  - USER INSTALLATION DIRECTORY: /Users/bedge/.gem/ruby/2.6.0
  - RUBY EXECUTABLE: /Users/bedge/.rbenv/versions/2.6.3/bin/ruby
  - GIT EXECUTABLE: /usr/local/bin/git
  - EXECUTABLE DIRECTORY: /Users/bedge/.rbenv/versions/2.6.3/bin
  - SPEC CACHE DIRECTORY: /Users/bedge/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /Users/bedge/.rbenv/versions/2.6.3/etc
  - RUBYGEMS PLATFORMS:
    - ruby
    - x86_64-darwin-18
  - GEM PATHS:
     - /Users/bedge/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0
     - /Users/bedge/.gem/ruby/2.6.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
     - "gem" => "--no-document"
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /Users/bedge/.rbenv/versions/2.6.3/bin
     - /usr/local/Cellar/rbenv/1.1.2/libexec
     - /usr/local/Cellar/zplug/2.4.2/bin
     - /Users/bedge/.zplug/bin
     - /usr/local/Cellar/pyenv-virtualenv/1.1.3/shims
     - /Users/bedge/.pyenv/shims
     - /Users/bedge/.jenv/shims
     - /Users/bedge/.bin
     - /usr/local/sbin
     - /usr/local/bin
     - /usr/local/opt/curl/bin
     - /usr/local/opt/gnu-sed/bin
     - /usr/local/opt/gnu-tar/libexec/gnubin
     - /usr/local/opt/coreutils/libexec/gnubin
     - /usr/bin
     - /bin
     - /usr/sbin
     - /sbin
```

--- a/lib/fpm/command.rb
+++ b/lib/fpm/command.rb
@@ -394,7 +394,12 @@ class FPM::Command < Clamp::Command
     set = proc do |object, attribute|
       # if the package's attribute is currently nil *or* the flag setting for this
       # attribute is non-default, use the value.
-      if object.send(attribute).nil? || send(attribute) != send("default_#{attribute}")
+
+      # Not all options have a default value, so we assume `nil` if there's no default. (#1543)
+      # In clamp >= 1.3.0, options without `:default => ..` will not have any # `default_xyz` 
+      # methods generated, so we need to check for the presence of this method first.
+      default = respond_to?("default_#{attribute}") ? send("default_#{attribute}") : nil
+      if object.send(attribute).nil? || send(attribute) != default
         logger.info("Setting from flags: #{attribute}=#{send(attribute)}")
         object.send("#{attribute}=", send(attribute))
       end

